synapse/lib/storm.py

```diff
@@ -2391,6 +2409,192 @@ async def execStormCmd(self, runt, genr):
         async for node, path in subg.run(runt, genr):
             yield node, path
 
+class ViewExecCmd(Cmd):
+    '''
+    Execute a storm query in a different view.
+
+    NOTE: Variables are passed through but nodes are not
+
+    Examples:
+
+        // Move some tagged nodes to another view
+        inet:fqdn#foo.bar $fqdn=$node.value() | view.exec 95d5f31f0fb414d2b00069d3b1ee64c6 { [ inet:fqdn=$fqdn ] }
+    '''
+
+    name = 'view.exec'
+    readonly = True
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('view', help='The GUID of the view in which the query will execute.')
+        pars.add_argument('storm', help='The storm query to execute on the view.')
+        return pars
+
+    async def execStormCmd(self, runt, genr):
+
+        # nodes may not pass across views, but their path vars may
+        node = None
+        async for node, path in genr:
+
+            view = await s_stormtypes.tostr(self.opts.view)
+            text = await s_stormtypes.tostr(self.opts.storm)
+
+            opts = {
+                'vars': path.vars,
+                'view': view,
+            }
+
+            query = await runt.getStormQuery(text)
+            async with runt.getSubRuntime(query, opts=opts) as subr:
+                async for item in subr.execute():
+                    await asyncio.sleep(0)
+
+            yield node, path
+
+        if node is None and self.runtsafe:
+            view = await s_stormtypes.tostr(self.opts.view)
+            text = await s_stormtypes.tostr(self.opts.storm)
+            query = await runt.getStormQuery(text)
+
+            opts = {'view': self.opts.view}
+            async with runt.getSubRuntime(query, opts=opts) as subr:
+                async for item in subr.execute():
+                    await asyncio.sleep(0)
+
+class BackgroundCmd(Cmd):
+    '''
+    Execute a query pipeline as a background task.
+    NOTE: Variables are passed through but nodes are not
+    '''
+    name = 'background'
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('query', help='The query to execute in the background.')
+        return pars
+
+    async def execStormTask(self, query, opts):
+
+        core = self.runt.snap.core
+        user = core._userFromOpts(opts)
+        info = {'query': str(query), 'opts': opts}
+
+        await core.boss.promote('storm', user=user, info=info)
```

Comment:
we'll need some way to know that a given task was backgrounded. possibly a flag in the info dict